factcast-store/src/test/java/org/factcast/store/internal/PgSynchronizedQueryTest.java

```diff
@@ -73,11 +82,53 @@ void testRunWithIndex() {
   void testRunWithoutIndex() {
     uut =
         new PgSynchronizedQuery(
-            jdbcTemplate, sql, setter, rowHandler, serialToContinueFrom, fetcher);
+            jdbcTemplate, sql, setter, rowHandler, serialToContinueFrom, fetcher, statementHolder);
     uut.run(false);
     verify(jdbcTemplate).execute(startsWith("SET LOCAL enable_bitmapscan"));
   }
 
+  @Test
+  @SneakyThrows
+  void test_exception_during_query() {
+    uut =
+        new PgSynchronizedQuery(
+            jdbcTemplate, sql, setter, rowHandler, serialToContinueFrom, fetcher, statementHolder);
+    when(statementHolder.wasCanceled()).thenReturn(false);
+    doThrow(DataAccessResourceFailureException.class)
+        .when(jdbcTemplate)
+        .query(anyString(), any(PreparedStatementSetter.class), any(RowCallbackHandler.class));
+
+    assertThatThrownBy(
```

Comment:
is this 

```java
        assertThatThrownBy(() -> uut.run(false)).isInstanceOf(DataAccessException.class);
```
too specific to assert here?

Reply:
changed it to assert the identity of the thrown exception, thx